### PR TITLE
[micromatch]: add options for scan method

### PR DESCRIPTION
**Pull request by mrmlnc**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/micromatch/picomatch/commit/c91f6ded31ed34808c218e3a661a973edc037ea0
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/micromatch/index.d.ts [hunk 1]
```diff
@@ -266,6 +266,60 @@ declare namespace micromatch {
...
+        /**
+         * When `true`, the returned object will include an array of strings representing each path "segment" in the scanned glob pattern.
+         * This is automatically enabled when `options.tokens` is `true`.
+         *
+         * @default false
+         */
+        parts?: boolean;
+    }
+
+    interface ScanInfo {
```
Comment (mrmlnc) [inline types/micromatch/index.d.ts]: JFYI

https://github.com/micromatch/picomatch/blob/66e1b200f3ea1784bdeae0770c8e5cd95f55e334/lib/scan.js#L319-L331

File: types/micromatch/index.d.ts [hunk 2]
```diff
@@ -266,6 +266,60 @@ declare namespace micromatch {
...
+        glob: string;
+        isBrace: boolean;
+        isBracket: boolean;
+        isGlob: boolean;
+        isExtglob: boolean;
+        isGlobstar: boolean;
+        negated: boolean;
+    }
+
+    interface ScanInfoToken {
```
Comment (mrmlnc) [inline types/micromatch/index.d.ts]: JFYI

https://github.com/micromatch/picomatch/blob/66e1b200f3ea1784bdeae0770c8e5cd95f55e334/lib/scan.js#L73

Other fields can be found by search: `token.`

File: types/micromatch/index.d.ts [hunk 3]
```diff
@@ -266,6 +266,60 @@ declare namespace micromatch {
...
+        isPrefix?: boolean;
+        negated?: boolean;
+    }
+
+    interface ScanInfoWithParts extends ScanInfo {
+        slashes: number[];
+        parts: string[];
+    }
+
+    interface ScanInfoWithTokens extends ScanInfoWithParts {
```
Comment (mrmlnc) [inline types/micromatch/index.d.ts]: JFYI

https://github.com/micromatch/picomatch/blob/66e1b200f3ea1784bdeae0770c8e5cd95f55e334/lib/scan.js#L333-L339

File: types/micromatch/index.d.ts [hunk 4]
```diff
@@ -266,6 +266,60 @@ declare namespace micromatch {
...
+        backslashes?: boolean;
+        isBrace?: boolean;
+        isBracket?: boolean;
+        isExtglob?: boolean;
+        isGlobstar?: boolean;
+        isPrefix?: boolean;
+        negated?: boolean;
+    }
+
+    interface ScanInfoWithParts extends ScanInfo {
```
Comment (mrmlnc) [inline types/micromatch/index.d.ts]: JFYI

https://github.com/micromatch/picomatch/blob/66e1b200f3ea1784bdeae0770c8e5cd95f55e334/lib/scan.js#L375-L376

**Review by vemoo — APPROVED** [review]

**Pull request merged**